### PR TITLE
#246 Add default APP_ENVIRONMENT

### LINKED ISSUE #246: Error on running app without APP_ENVIRONMENT
If `APP_ENVIRONMENT` is not set, the following error is thrown:

```
thread 'main' panicked at 'Failed to parse configuration settings: NotPresent', src/main.rs:156:44
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### PR DESCRIPTION
**Pull request by wlthomson**
Closes #246.

### CODE REVIEW

**Comment by andreievg** [comment]
readme too ?

**Comment by wlthomson** [comment]
> 
> 
> readme too ?

9076a10.

File: src/util/configuration.rs [hunk 1]
```diff
@@ -35,8 +35,7 @@ pub fn get_configuration_base_file() -> Result<File<FileSourceFile>, SettingsErr
 /// environments are `local` and `production`.
 pub fn get_configuration_app_file() -> Result<File<FileSourceFile>, SettingsError> {
     let configuration_directory = get_configuration_directory()?;
-    let app_file =
-        File::from(configuration_directory.join(AppEnvironment::try_get()?)).required(true);
+    let app_file = File::from(configuration_directory.join(AppEnvironment::get())).required(true);
```
Comment (clemens-msupply) [inline src/util/configuration.rs:38]: Unrelated, why is there a try_get() method? seems to be not needed?

**Review by clemens-msupply — APPROVED** [review]

File: src/util/configuration.rs (hunk 1 above)
Comment (wlthomson) [reply src/util/configuration.rs:38]: Tbh, it's not really needed anymore. I left it in as an example of a "pattern" that I like and think would be good to follow throughout the codebase (I've noticed it's used by a lot of rust crates). It goes completely against the laws of YAGNI, but I think it's fine to leave in there and exposes a nice interface (I could see the configuration stuff all being modularised as a crate at some point).

Comment (wlthomson) [reply src/util/configuration.rs:38]: Going to go ahead and merge (if anyone strongly opposes the `try_get()` being included, I won't protest if they remove it).

**Pull request merged**